lib/sharing/ShareRuntimeModule.js

```diff
@@ -83,11 +83,11 @@ class ShareRuntimeModule extends RuntimeModule {
 					)};`,
 					`var uniqueName = ${JSON.stringify(uniqueName || undefined)};`,
 					`var register = ${runtimeTemplate.basicFunction(
-						"name, version, factory",
+						"name, version, factory, eager",
 						[
 							"var versions = scope[name] = scope[name] || {};",
 							"var activeVersion = versions[version];",
-							"if(!activeVersion || !activeVersion.loaded && uniqueName > activeVersion.from) versions[version] = { get: factory, from: uniqueName };"
+							"if(!activeVersion || (!activeVersion.loaded && !activeVersion.eager && (uniqueName > activeVersion.from || eager))) versions[version] = { get: factory, from: uniqueName, eager: eager };"
```

Comment:
```suggestion
							"if(!activeVersion || (!activeVersion.loaded && (eager == !activeVersion.eager ? eager : uniqueName > activeVersion.from)) versions[version] = { get: factory, from: uniqueName, eager: eager };"
```

Otherwise the case where `activeVersion.eager === eager === true` won't be handled correctly.

Reply:
Thanks for the feedback.  Committed that change and added tests to make sure the name comparison still works when activeVersion.eager === eager.